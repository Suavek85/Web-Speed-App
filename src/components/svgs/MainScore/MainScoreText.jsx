import React from "react"
import { useSpring, animated } from 'react-spring'

export default function MainScoreText(props) {
  const { xCord, yCord, value } = props

  const styles = useSpring({  
    from: { number: 0 }, 
    to: { number: value },
    config: { duration: 2000 },
  })

  return (
    <animated.text 
      x={xCord}
      y={yCord}
      fill="darkslategrey" 
      fontWeight='bold'
      fontFamily='Orbitron, sans-serif' 
      fontSize="65" > 
      {value && styles.number.interpolate(number => Math.floor(number))}
    </animated.text>
  )
}


