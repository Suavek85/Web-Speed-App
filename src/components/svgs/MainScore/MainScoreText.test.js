import React from 'react'
import { shallow } from 'enzyme'
import MainScoreText from './MainScoreText'

it('should match the snapshot', () => {
  const wrapper = shallow(<MainScoreText />)
  expect(wrapper).toMatchSnapshot()
})