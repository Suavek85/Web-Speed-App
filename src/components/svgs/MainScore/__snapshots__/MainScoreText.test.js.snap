// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<ForwardRef
  fill="darkslategrey"
  fontFamily="Orbitron, sans-serif"
  fontSize="65"
  fontWeight="bold"
/>
`;
